js/modules/k6/html/html.go

```diff
@@ -71,3 +99,438 @@ func (s Selection) Attr(name string, def ...goja.Value) goja.Value {
 	}
 	return s.rt.ToValue(val)
 }
+
+func (s Selection) Html() goja.Value {
+	val, err := s.sel.Html()
+	if err != nil {
+		return goja.Undefined()
+	}
+	return s.rt.ToValue(val)
+}
+
+func optionVal(s *goquery.Selection) string {
+	if val, exists := s.Attr("value"); exists {
+		return val
+	}
+
+	if val, err := s.Html(); err == nil {
+		return val
+	}
+
+	return ""
+}
+
+func (s Selection) Val() goja.Value {
+	switch goquery.NodeName(s.sel) {
+	case "input":
+		return s.Attr("value")
+
+	case "textarea":
+		return s.Html()
+
+	case "button":
+		return s.Attr("value")
+
+	case "option":
+		return s.rt.ToValue(optionVal(s.sel))
+
+	case "select":
+		selected := s.sel.First().Find("option[selected]")
+
+		if _, exists := s.sel.Attr("multiple"); exists {
+			return s.rt.ToValue(selected.Map(func(idx int, opt *goquery.Selection) string { return optionVal(opt) }))
+		} else {
+			return s.rt.ToValue(optionVal(selected))
+		}
+
+	default:
+		return goja.Undefined()
+	}
+}
+
+func (s Selection) Closest(arg goja.Value) Selection {
+	return s.varargFnCall(arg, s.sel.Closest, s.sel.ClosestSelection, s.sel.ClosestNodes)
+}
+
+func (s Selection) Children(def ...string) Selection {
+	if len(def) == 0 {
+		return Selection{s.rt, s.sel.Children()}
+	} else {
+		return Selection{s.rt, s.sel.ChildrenFiltered(def[0])}
+	}
+}
+
+func (s Selection) Contents() Selection {
+	return Selection{s.rt, s.sel.Contents()}
+}
+
+func (s Selection) Each(v goja.Value) Selection {
+	gojaFn, isFn := goja.AssertFunction(v)
+	if isFn {
+		fn := func(idx int, sel *goquery.Selection) {
+			gojaFn(v, s.rt.ToValue(idx), selToElement(s))
+		}
+
+		return Selection{s.rt, s.sel.Each(fn)}
+	} else {
+		panic(s.rt.NewGoError(errors.New("Argument to each() must be a function")))
+		return s
+	}
+}
+
+func (s Selection) End() Selection {
+	return Selection{s.rt, s.sel.End()}
+}
+
+func (s Selection) buildMatcher(v goja.Value, gojaFn goja.Callable) func(int, *goquery.Selection) bool {
+	return func(idx int, sel *goquery.Selection) bool {
+		fnRes, fnErr := gojaFn(v, s.rt.ToValue(idx), s.rt.ToValue(sel))
+		return fnErr == nil && fnRes.ToBoolean()
+	}
+}
+
+func (s Selection) Filter(v goja.Value) Selection {
+	if gojaFn, isFn := goja.AssertFunction(v); isFn {
```

Comment:
I couldn't find a variation of gojaFn, isFn := goja.AssertFunction(v) that worked in a case statement, am I missing something? 

Ended up using an inlined type switch and moving the AssertFunction outside the switch - it looks less cramped but I'm not sure it's idiomatic.